ykcompile/src/lib.rs

```diff
@@ -178,8 +191,10 @@ impl<TT> TraceCompiler<TT> {
                 } else {
                     // All registers are occupied, so we need to spill the local to the stack. For
                     // now we assume that all spilled locals are 8 bytes big.
-                    let loc = Location::Stack(self.stack_builder.alloc(8, 8) as i32);
-                    loc
+                    Location::new_mem(
+                        RBP.code(),
+                        -i32::try_from(self.stack_builder.alloc(8, 8)).unwrap(),
```

Comment:
Maybe writing this as `-1 * i32::...` makes it more difficult to overlook the `-` here.

Reply:
We should also change `stack_builder` to return an i32, but that's for another PR.

Reply:
That being said, it does return `u32`. Shouldn't it be safe to cast to `i32` using `as` in this case?

Reply:
The stack builder now returns a Location directly. See the second commit.

Reply:
Should we still make that change there? At least now it's only one place to change. ;-)

Reply:
Do you mean, why don't we use `as` in the stackbuilder when creating a location? Instead of `try_from`?

```rust
Location::new_mem(RBP.code(), -i32::try_from(self.stack_top).unwrap())
```

Using `as` wouldn't be safe, as an i32 can't express all of the numbers `stack_top` can (it's a `u64`).

Generally, I try to avoid `as` unless really necessary. Using `try_from` costs nothing when there can be no error, and it protects us, should the type change behind our backs.

Reply:
I meant adding the `-1 * i32::...`, otherwise the `-` can be easily overlooked (I barely noticed it was there the first time I saw this code).

Reply:
Fixed.